app/components/UI/Ramp/common/hooks/useBalance.ts

```diff
@@ -0,0 +1,66 @@
+import { hexToBN } from '@metamask/controller-utils';
+import { useSelector } from 'react-redux';
+import { NATIVE_ADDRESS } from '../../../../../constants/on-ramp';
+import { selectAccounts } from '../../../../../selectors/accountTrackerController';
+import {
+  selectConversionRate,
+  selectCurrentCurrency,
+} from '../../../../../selectors/currencyRateController';
+import { selectSelectedAddress } from '../../../../../selectors/preferencesController';
+import { selectContractBalances } from '../../../../../selectors/tokenBalancesController';
+import { selectContractExchangeRates } from '../../../../../selectors/tokenRatesController';
+import { safeToChecksumAddress } from '../../../../../util/address';
+import {
+  balanceToFiat,
+  renderFromTokenMinimalUnit,
+  renderFromWei,
+  weiToFiat,
+} from '../../../../../util/number';
+
+interface Asset {
+  address: string;
+  decimals: number;
+}
+
+export default function useBalance(asset?: Asset) {
+  const assetAddress = safeToChecksumAddress(asset?.address);
+  const accounts = useSelector(selectAccounts);
+  const selectedAddress = useSelector(selectSelectedAddress);
+  const conversionRate = useSelector(selectConversionRate);
+  const currentCurrency = useSelector(selectCurrentCurrency);
+  const tokenExchangeRates = useSelector(selectContractExchangeRates);
+  const balances = useSelector(selectContractBalances);
+
+  if (!asset || !assetAddress) {
+    return { balance: null, balanceFiat: null, balanceBN: null };
+  }
+
+  let balance, balanceFiat, balanceBN;
+  if (assetAddress === NATIVE_ADDRESS) {
+    balance = renderFromWei(accounts[selectedAddress]?.balance);
+    balanceBN = hexToBN(accounts[selectedAddress].balance);
+    balanceFiat = weiToFiat(balanceBN, conversionRate, currentCurrency);
+  } else {
+    const exchangeRate =
+      assetAddress && assetAddress in tokenExchangeRates
+        ? tokenExchangeRates[assetAddress]
+        : undefined;
+    balance =
+      assetAddress && assetAddress in balances
+        ? renderFromTokenMinimalUnit(
+            balances[assetAddress],
+            asset.decimals ?? 18,
+          )
+        : 0;
```

Comment:
Suggestion: try to avoid [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) even if they make perfect sense for some seasoned crypto dev, consider someone who discovers the code could wonder why 18 or why returning 0 by default. It doesn't change the logic, only makes it easier to read code like "plain english".

```suggestion
            asset.decimals ?? DEFAULT_ETHEREUM_DECIMALS_LENGTH,
          )
        : EMPTY_BALANCE;
```

Reply:
Great feedback! will address in a upcoming PR!